spec/support/test_access_patch.rb

```diff
@@ -1,30 +0,0 @@
-# typed: true
```

Comment:
We have this quite clever monkey patch to expose private members for testing, but is actually never used.

This kind of test-specific, internal access to the tracer (specifically monitoring if the `AsyncTransport` has finished flushing) was the best way to write reliable tests around HTTP connection reuse.

**But polluting the main class space with methods that won't actually be present when users use `ddtrace` in production is scary, since we could write features that use methods that are not available in production, but will always succeed in our test suite.**

Because of this, I've moved this monkey patching to a refinement, so that it only applies to the current spec file under test, never leaking to `lib/` code.

Because the file changed so much, git thinks I completely deleted it (which I kind of did, for most of the unused features). `spec/support/test_access_refinement.rb` contains these new refinements.